tests/components/group/test_init.py

```diff
@@ -18,307 +17,375 @@
-
-        self.hass.states.set("light.Bowl", STATE_ON)
-        self.hass.block_till_done()
-
-        state = self.hass.states.get(test_group.entity_id)
-        assert not state.attributes.get(ATTR_ASSUMED_STATE)
-
-    def test_group_updated_after_device_tracker_zone_change(self):
-        """Test group state when device tracker in group changes zone."""
-        self.hass.states.set("device_tracker.Adam", STATE_HOME)
-        self.hass.states.set("device_tracker.Eve", STATE_NOT_HOME)
-        self.hass.block_till_done()
-        group.Group.create_group(
-            self.hass, "peeps", ["device_tracker.Adam", "device_tracker.Eve"]
-        )
-        self.hass.states.set("device_tracker.Adam", "cool_state_not_home")
-        self.hass.block_till_done()
-        assert STATE_NOT_HOME == self.hass.states.get(f"{group.DOMAIN}.peeps").state
+async def test_setup_group_with_mixed_groupable_states(hass):
+    """Try to set up a group with mixed groupable states."""
+
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("device_tracker.Paulus", STATE_HOME)
+
+    assert await async_setup_component(hass, "group", {})
+
+    await group.Group.async_create_group(
+        hass, "person_and_light", ["light.Bowl", "device_tracker.Paulus"]
+    )
+
+    await hass.async_block_till_done()
+
+    assert STATE_ON == hass.states.get(f"{group.DOMAIN}.person_and_light").state
+
+
+async def test_setup_group_with_a_non_existing_state(hass):
+    """Try to set up a group with a non existing state."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+
+    assert await async_setup_component(hass, "group", {})
+
+    grp = await group.Group.async_create_group(
+        hass, "light_and_nothing", ["light.Bowl", "non.existing"]
+    )
+
+    assert STATE_ON == grp.state
+
+
+async def test_setup_group_with_non_groupable_states(hass):
+    """Test setup with groups which are not groupable."""
+    hass.states.async_set("cast.living_room", "Plex")
+    hass.states.async_set("cast.bedroom", "Netflix")
+
+    assert await async_setup_component(hass, "group", {})
+
+    grp = await group.Group.async_create_group(
+        hass, "chromecasts", ["cast.living_room", "cast.bedroom"]
+    )
+
+    assert grp.state is None
+
+
+async def test_setup_empty_group(hass):
+    """Try to set up an empty group."""
+    grp = await group.Group.async_create_group(hass, "nothing", [])
+
+    assert grp.state is None
+
+
+async def test_monitor_group(hass):
+    """Test if the group keeps track of states."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("light.Ceiling", STATE_OFF)
+
+    assert await async_setup_component(hass, "group", {})
+
+    test_group = await group.Group.async_create_group(
+        hass, "init_group", ["light.Bowl", "light.Ceiling"], False
+    )
+
+    # Test if group setup in our init mode is ok
+    assert test_group.entity_id in hass.states.async_entity_ids()
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)